src/main.rs

```diff
@@ -215,8 +215,9 @@ async fn process_feed(
         .url
         .parse()
         .wrap_err_with(|| format!("unable to parse {} as a URL", config.url))?;
-    let req = client.get(url.clone());
-    let req = maybe_add_cache_headers(req, cached_headers);
+    let mut req = client.get(url.clone());
+    req = add_headers(req, cached_headers, &channel_config.user_agent);
```

Comment:
Think I preferred the prior construct where `req` was shadowed and therefore didn't need to be mutable. That would look like this:

```suggestion
    let req = client.get(url.clone());
    let req = add_headers(req, cached_headers, &channel_config.user_agent);
```

Better yet might be:

```suggestion
    let req = add_headers(client.get(url.clone()), cached_headers, &channel_config.user_agent);
```

Reply:
Sounds good